policysync/processor_test.go

```diff
@@ -640,19 +836,74 @@ func deltaUpdateIpSet(id string, add, del int) *proto.IPSetDeltaUpdate {
 		Id: id,
 	}
 	for i := 0; i < add; i++ {
-		msg.AddedMembers = append(msg.AddedMembers, makeIP(i))
+		msg.AddedMembers = append(msg.AddedMembers, makeIPAndPort(i))
 	}
 	for i := add; i < add+del; i++ {
-		msg.RemovedMembers = append(msg.RemovedMembers, makeIP(i))
+		msg.RemovedMembers = append(msg.RemovedMembers, makeIPAndPort(i))
 	}
 	return msg
 }
 
-func makeIP(i int) string {
-	o := make([]string, 4)
-	o[0] = strconv.Itoa(i & 0xff000000 >> 24)
-	o[1] = strconv.Itoa(i & 0x00ff0000 >> 16)
-	o[2] = strconv.Itoa(i & 0x0000ff00 >> 8)
-	o[3] = strconv.Itoa(i & 0x000000ff)
-	return strings.Join(o, ".")
+func makeIPAndPort(i int) string {
+	// Goal here is to make the IPSet members as long as possible when stringified.
+	// assume 20 bits of variable and 108 bits of fixed prefix
+	lsbHex := fmt.Sprintf("%05x", i)
+
+	return "fe80:1111:2222:3333:4444:5555:666" + string(lsbHex[0]) + ":" + lsbHex[1:] + ",tcp:65535"
+}
+
+func getDialOptions() []grpc.DialOption {
+	return []grpc.DialOption{
+		grpc.WithInsecure(),
+		grpc.WithDialer(getDialer("unix"))}
+}
+
+func getDialer(proto string) func(string, time.Duration) (net.Conn, error) {
+	return func(target string, timeout time.Duration) (net.Conn, error) {
+		return net.DialTimeout(proto, target, timeout)
+	}
+}
+
+const ListenerDir = "/tmp/felixut"
```

Comment:
It'd be best to use a per-test temp dir to avoid any cross-contamination if tests are run in parallel (or if any garbage is left from a previous failure).